resources/test_data/sqlite_testrunner_queries.sql

```diff
@@ -191,7 +191,8 @@ SELECT * FROM id_int_int_int_100 AS r WHERE a < (SELECT SUM(min_a) FROM (SELECT
 SELECT a, SUM(b) FROM mixed GROUP BY a;
 SELECT a, SUM(b), AVG(c) FROM mixed GROUP BY a;
 SELECT a, b, MAX(c), AVG(b) FROM mixed GROUP BY a, b;
-SELECT a AS whatever, SUM(b) FROM mixed GROUP BY whatever;
+SELECT a AS a_alias, SUM(b) FROM mixed GROUP BY a_alias;
```

Comment:
Some more queries would be justified here, imo. What about the aliases being used in WHERE? in HAVING? is it legal to use them there?

Reply:
I'd also add tests that make sure that the aliases properly survive joins etc.

Reply:
Do we want to support queries like `SELECT a AS x FROM mixed WHERE x > 10`? The `SQLTranslator` does not support this at the moment.

Reply:
I believe that this is inconsistent across other DBMSs. My thoughts are:

1. If we need it for TPC-DS and it is reasonably doable, do it.
2. If it's not part of the TPC-DS syntax but can be tacked on without significant extra effort, do it.
3. If adding it adds significant effort, don't.
3a. If possible, add new assertions so that we, should it ever come around, get a more helpful message (maybe even with a comment outlining what would have to be done)

Reply:
I imagine this requires effort and agree with Markus' other points

Reply:
I was able to add alias support in the WHERE clause (https://github.com/hyrise-mp/hyrise/commit/6d9c7e8b7a642ba97a2cd845914ea773911f2bf5). However, it has some consequences:

Reply:
>     2. If it's not part of the TPC-DS syntax but can be tacked on without significant extra effort, do it.
Yes, it's not required for the TPC-DS.

Reply:
I don't see a clear win here and it breaks even more with this comment, though I wouldn't see this as important unless the referenced blog post would actually reference the standard.

```
  // SQL Orders of Operations: http://www.bennadel.com/blog/70-sql-query-order-of-operations.htm
  // 1. FROM clause (incl. JOINs and sub-SELECTs that are part of this)
  // 2. WHERE clause
  // 3. GROUP BY clause
  // 4. HAVING clause
  // 5. SELECT clause (incl. DISTINCT)
  // 6. UNION clause
  // 7. ORDER BY clause
  // 8. LIMIT clause
```

We gain one thing and loose another - I'm fine with it. 

> This does not seem to be a problem from my perspective. I only wondered because there is an explicit test for this

Don't worry, the ParamaterIDs are arbitrary, they just have to be unique. `parameter_ids_of_value_placeholders` is what gives them order. Explicit values were tested to keep things simple.

Unless @mrks says something else, I'd say we can keep this. Please revise the naming of `_translate_select_list_groupby_having` as its behavior has  changed. Adjust that comment I mentioned above (probably removing the URL)

Reply:
Sounds good.

> [...] not by Hyrise anymore because a becomes ambiguous.

So be it. Whoever writes those queries deserves to be punished. ;)